test/java/seedu/addressbook/commands/UndoDeleteCommandTest.java

```diff
@@ -0,0 +1,157 @@
+
+import org.junit.Before;
+import org.junit.Test;
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.*;
+import seedu.addressbook.data.person.UniquePersonList.PersonNotFoundException;
+import seedu.addressbook.data.tag.UniqueTagList;
+import seedu.addressbook.ui.TextUi;
+import seedu.addressbook.util.TestUtil;
+import seedu.addressbook.data.PreviousDeletes;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class UndoDeleteCommandTest {
+
+    private AddressBook emptyAddressBook;
+    private AddressBook addressBook;
+
+    private List<ReadOnlyPerson> emptyDisplayList;
+    private List<ReadOnlyPerson> listWithEveryone;
+    private List<ReadOnlyPerson> listWithSurnameDoe;
+    private PreviousDeletes deletes;
+
+    @Before
+    public void setUp() throws Exception {
+        Person johnDoe = new Person(new Name("John Doe"), new Phone("61234567", false),
+                new Email("john@doe.com", false), new Address("395C Ben Road", false), new UniqueTagList());
+        Person janeDoe = new Person(new Name("Jane Doe"), new Phone("91234567", false),
+                new Email("jane@doe.com", false), new Address("33G Ohm Road", false), new UniqueTagList());
+        Person samDoe = new Person(new Name("Sam Doe"), new Phone("63345566", false),
+                new Email("sam@doe.com", false), new Address("55G Abc Road", false), new UniqueTagList());
+        Person davidGrant = new Person(new Name("David Grant"), new Phone("61121122", false),
+                new Email("david@grant.com", false), new Address("44H Define Road", false),
+                new UniqueTagList());
+
+        emptyAddressBook = TestUtil.createAddressBook();
+        addressBook = TestUtil.createAddressBook(johnDoe, janeDoe, davidGrant, samDoe);
+
+        emptyDisplayList = TestUtil.createList();
+        deletes = new PreviousDeletes();
+
+        listWithEveryone = TestUtil.createList(johnDoe, janeDoe, davidGrant, samDoe);
+        listWithSurnameDoe = TestUtil.createList(johnDoe, janeDoe, samDoe);
+    }
+
+    @Test
+    public void execute_successful_undoDelete() throws PersonNotFoundException, UniquePersonList.DuplicatePersonException {
+        assertUndoDeletionSuccessful(1, addressBook, listWithSurnameDoe);
+    }
+
+    /**
+     * Creates a new delete command.
+     *
+     * @param targetVisibleIndex of the person that we want to delete
+     */
+    private DeleteCommand createDeleteCommand(int targetVisibleIndex, AddressBook addressBook,
+                                                                      List<ReadOnlyPerson> displayList) {
+
+        DeleteCommand command = new DeleteCommand(targetVisibleIndex);
+        command.setData(addressBook, displayList);
+
+        return command;
+    }
+
+    /**
+     * Creates a new undoDelete command.
+     *
+     * @param
```

Comment:
Remove these empty param and return strings if it is obvious from the method signature.